src/HexagonOptimize.cpp

```diff
@@ -171,6 +171,80 @@ Expr wild_i16x = Variable::make(Type(Type::Int, 16, 0), "*");
 Expr wild_i32x = Variable::make(Type(Type::Int, 32, 0), "*");
 Expr wild_i64x = Variable::make(Type(Type::Int, 64, 0), "*");
 
+// Check if a pattern with flags 'flags' is supported on the target.
+bool check_pattern_target(int flags, const Target &target) {
+    if ((flags & (Pattern::v62orLater)) &&
+        !target.features_any_of({Target::HVX_v62, Target::HVX_v65, Target::HVX_v66})) {
+        return false;
+    }
+    if ((flags & (Pattern::v65orLater)) &&
```

Comment:
I think we should do this in a separate change, right now this is just code motion, and because it involves changing a lot of plumbing changes (passing Target -> something else).

Reply:
Indeed yes, it should be a separate change. Main intention was to get your thoughts on the idea. Would like to push it into #2913 ideally.

Reply:
I like the idea, and I was thinking, we could do it without any plumbing changes, just make a helper function int get_max_hvx_version(const Target&) and just use it on-site.